### ISSUE TITLE
Steam client re-download gigabytes of game updates almost every day but in reality it is no game update.

### ISSUE THREAD

**Issue by drmmwlt** [ValveSoftware/steam-for-linux#10285]
#### Your system information

* Steam client version (build number or date): 1701289036
* Distribution (e.g. Ubuntu): Manjaro
* Opted into Steam client beta?: [Yes/No] no
* Have you checked for system updates?: [Yes/No] yes
* Steam Logs: [generate by running this command in a terminal `tar -zcvf ~/Desktop/steam-logs.tar.gz ~/.steam/steam/logs`]

(not working)

tar: Suppression de « / » au début des noms des membres
tar (child): /home/fleuve/Desktop/steam-logs.tar.gz : open impossible: Aucun fichier ou dossier de ce type
tar (child): Error is not recoverable: exiting now
/home/fleuve/.steam/steam/logs/
/home/fleuve/.steam/steam/logs/controller_ui.txt
tar: /home/fleuve/Desktop/steam-logs.tar.gz : write impossible: Relais brisé (pipe)
tar: Child returned status 2
tar: Error is not recoverable: exiting now

* GPU: Nvidia

#### Please describe your issue in as much detail as possible:

(same issue as #9174 but was closed)

Steam downloads gibytes of updates when there is no update. Since 3 days steam is downloading the exact same game updates, (here Dota 2, AeO 2, GTA5, Steep) , same size. This kind of of loop happens regularly on this computer even on fresh system installs. I would like to download game updates only 1 time.

#### Steps for reproducing this issue:

1.  Open Steam, Updates games
2.  play few hours, shutdown steam and the computer
3.  next day, open steam

I tried clear download cache, deleting the inside of /home/.steam/root/package , sometimes it works, sometimes it doesnt. Here steam is still stuck in the loop.

In #9174, @kisak-valve mentions the shader_log.txt /home/fleuve/.local/share/Steam/logs so here it is. 

[shader_log.txt](https://github.com/ValveSoftware/steam-for-linux/files/13626843/shader_log.txt)

**Comment by drmmwlt** [comment]
damn it's happening even after quitting the game 

![Capture d’écran du 2023-12-12 09-09-31](https://github.com/ValveSoftware/steam-for-linux/assets/66220502/e175bea7-021a-4f8b-b487-8e3cb351f090)

**Comment by shadowscion** [comment]
I set them all to update on launch so at least all my games aren't updating. It seems for now the only options are to deal with it, sleep while plugged in, or pointlessly drain the battery.

**Comment by hjpaul7** [comment]
I also have this, on desktop and Steam Deck. Nearly every day there's almost _200_ updates. Assuming shader cache?

![steam-downloads](https://github.com/ValveSoftware/steam-for-linux/assets/63761039/15c0436c-b01e-4fae-a63e-e05b5321c979)

**Comment by drmmwlt** [comment]
not a fix but i tried the steam flatpak version for few days. It works flawlessly imo

**Comment by Kopetan4egX** [comment]
Same problem. Up to 1GB+ updates every day. Sometimes even after every restart of the steam client.

**Comment by drmmwlt** [comment]
day 4 of redownloading every game updates everydays

![Capture d’écran du 2024-01-11 17-19-13](https://github.com/ValveSoftware/steam-for-linux/assets/66220502/de909180-6cd0-41ea-8633-cab1a442051a)

**Comment by shadowscion** [comment]
> I set them all to update on launch so at least all my games aren't updating every boot. Seems to be at least a 2 year old problem.

The issue has returned.

**Comment by Brejvinder** [comment]
Experiencing the same issue, 5.8/5.9GB update for Warframe and other games, Ghostrunner 1&2, Titanfall 2 , etc. every time. Pretty much on every steam restart without updating my system or anything else.

My details:

OS: EndeavourOS Linux x86_64
Kernel: 6.6.10-zen1-1-zen
Steam Beta Branch:  Stable Client
Steam Version:  1705108172
Steam Client Build Date:  Fri, Jan 12 7:52 PM UTC -08:00
Steam Web Build Date:  Fri, Jan 12 12:02 PM UTC -08:00
Steam API Version:  SteamClient021

![download issue](https://github.com/ValveSoftware/steam-for-linux/assets/33681708/aff0cb32-a311-4148-9662-c16e6dd392c8)

**Comment by RoanBrand** [comment]
Happening to me too. Maybe because 2nd added steam library is on exFat partition with its respective compatdata and shader folder?
I know proton doesn't like exFat due to symlinks missing, so I have that on main / steam lib on ext4

**Comment by eode** [comment]
Shouldn't be that, because I'm on one typical (but BTRFS) setup of / and `/home`, with all games in `/home/<username>`.  I'm running into this same issue.

**Comment by drmmwlt** [comment]
everyday everyday.... devs, how can we help you fixing this ?
![Capture d’écran du 2024-01-23 00-09-44](https://github.com/ValveSoftware/steam-for-linux/assets/66220502/176da1e3-a4d7-46a6-9536-d2277275df3a)

**Comment by eode** [comment]
I have a separate issue (at least, separate as far as I know) that causes games to *validate* every day as well.  These issues both being present compounds the difficulty.

**Comment by H3r4cles** [comment]
It seems that pausing the download when u first launch steam then clear all the downloads fixes this issue for me but only if i don't restart steam.
It is also a good way to figure out if its an actual game content/patch download and not a steam pre cache update. Because if it was really a patch for a game then it will show up as scheduled for download after you cleared all your downloads.
Maybe making a fix around this should solve this problem.

**Comment by eode** [comment]
That doesn't work for me.  Every time I quit the game or restart steam, it re-downloads shaders (for some games in particular).

**Comment by Soulprayer79** [comment]
At the moment I have 9 games installed, three of them are linux native games (X4:Foundations, Factorio and Dwarf Fortress)
I need to download it every day for about half the games, but the biggies are:
Linux-native: X4F (~500MB), 
Proton games: Baldur's Gate 3 (~450MB) and Enshrouded (8 **Giga**Byte) 
[steam-logs.tar.gz](https://github.com/ValveSoftware/steam-for-linux/files/14083821/steam-logs.tar.gz)

**Comment by sophie-kuehn** [comment]
Never saw this an an issue. But now i installed Enshrouded and get a 8GB shader-cache update every time i start steam or exit the game. 8GB. That also take ages processing when i want to start the game. Is there any workaround other than disabling shader pre-caching for all games?

**Comment by H3r4cles** [comment]
@sophie-kuehn pausing and clearing the downloads makes this shader cache update go away as long as i don't restart steam.
This is the only way i have found until valve figure it out.

**Comment by sophie-kuehn** [comment]
@H3r4cles This is not a solution for me as i restart the computer daily. What is my temporary solution now is to set a update schedule between 5 and 6 in the morning when the pc is never running. And therefore prevent all updates without removing the shader cache. I now need to plan a weekly update day for myself i suppose ...

**Comment by Zorrototo** [comment]
The daily download is back I can confirm. I didn't find the old thread so I reply in this newer one.

**Comment by arcangel2p** [comment]
The same happens on Steam Deck (game and desktop modes). At lesast on Preview and Main channels.

**Comment by Rosentti** [comment]
Are these game updates, shader updates or workshop content? You can put your mouse over the "i" symbol next to the currently downloading app, and it'll show what it's downloading. I also sometimes get empty downloads, for which Steam doesn't show any info, including sizes, patchnotes or the "i" symbol.

**Comment by sophie-kuehn** [comment]
> Are these game updates, shader updates or workshop content? You can put your mouse over the "i" symbol next to the currently downloading app, and it'll show what it's downloading. I also sometimes get empty downloads, for which Steam doesn't show any info, including sizes, patchnotes or the "i" symbol.

These are shader updates. Sometimes pared with content updates.

**Comment by shelterx** [comment]
I had to disable shader caching for this to stop, at first i thought it was game updates because steam isn't clear on what it's actually downloading.

**Comment by emilstark** [comment]
Happens everyday for Steam + Linux, ~1GB DOTA update after each restart.

**Comment by Zorrototo** [comment]
The worst to troubleshoot is that it is not consistent. You start Steam it downloads shader cache update for a dozen game, totaling 1GB+ download. You REBOOT the system, your start Steam, it does the same again. You reboot again, it does the same, you reboot again, it does not download anything. There is no logic at all.

I'm on Steam BETA but it does the same on Steam non beta currently.

**Comment by beach-89** [comment]
This is still happening almost every reboot like the comment above says. It will download the entire shader cache, then redownload it 5 min later after a restart. Is there any logs or anything that can be provided to help elevate or diagnose this?

**Comment by fireba11** [comment]
That's been an issue for *months* now ... still no fixes?
Apparently download traffic so too cheap now :-P

**Comment by Zorrototo** [comment]
It only it was 50MB per games, but I have this 2.7GB Rocket League shader cache update every time almost, this is getting old, I only have 10Mb DSL internet, it takes hours to finish it all, and if I need to reboot for whatever reason, I'm up for another round of download...

**Comment by Daehdam** [comment]
Hello mates,
i haven´t fixed that Problem jet but i have the same and i think i am on the edge to understand the Problem...
It seems to be the "Shader Cache" witch is - as far as i get that - some individual computerinformations for your Grafikcard so the game runs smooth - Problem is the default sice is too damn small so the Game has to reload the missing parts --> The solution i found so far is to set the default sice up from 4GB to 10GB or even higher (My Shader for Enshroudet needs 17GB :O )

I am looking for an easy way to do that but atm i just know what to do but not how xD i found this post do - maybe one of you can fix their Problems with this information (tell me how to do it then pls)

 It has been suggested that setting the following flags in your profile (i.e. inside the file ~/.profile or similar) may help with the bloated shader cache that most recent Nvidia drivers seem to produce.

export __GL_SHADER_DISK_CACHE_SKIP_CLEANUP=1
export __GL_SHADER_DISK_CACHE_SIZE=100000000000 #100 gb max disk size of cache

**Comment by Zorrototo** [comment]
Isn't this completely unrelated to the Steam shader cache files downloaded in Steam?

**Comment by DimitarSamarov07** [comment]
I can confirm this is an issue. Warframe downloads 7.3GB of Shader pre-cache data after every steam/PC reboot. This has to be fixed!

**Comment by shadowscion** [comment]
> I can confirm this is an issue. Warframe downloads 7.3GB of Shader pre-cache data after every steam/PC reboot. This has to be fixed! System: Manjaro with AMD GPU and CPU

That game is what made me disable the pre-caching feature entirely. And you know what? I have not noticed one bit of difference (on my Steam Deck).

**Comment by TDYK** [comment]
@ValveSoftware Hi, any update/responce?
Tell us what logs/system reports to send you for debugging. Yesterday i downloaded some gigas of waste, today here is the screenshot:
![still](https://github.com/ValveSoftware/steam-for-linux/assets/3619560/062ae2f3-8911-4a0a-a731-fe25bac2e6a5)
Please.

**Comment by sophie-kuehn** [comment]
Is Valve actually monitoring this channel? The lack of official feedback is unsettling.

**Comment by Zorrototo** [comment]
I guess if we were asked "provide this info, this info, describe this or that" it would speed up the process, I could give random info/log but I don't know what is needed to troubleshoot the issue on Valve's end.

**Comment by shelterx** [comment]
This must be quite easy to replicate.
I use steam native with nothing fancy stuff going on. I have a second games library pointing to my ntfs drive. I have symlinks for the compatdata and shader folders, that's the only "special" thing I have.

**Comment by uunnxx** [comment]
Hi there!
Any updates on this ?


I'm getting tired of this, every time, when I quite game, it has to validate.
Only and only in few cases it'll get less than 1GB, in the other cases, I've to get more than 1GB, in case I quite the game.


![image](https://github.com/ValveSoftware/steam-for-linux/assets/6236057/685b3db9-0114-40e3-961f-758f8b1256af)

**Comment by ettige** [comment]
I haveing this problem mainly on Valv's games like CS2, it very annoying for me i need to download 2GB of shader and still need to compile it on my own to do not feel any lag, it would be better to be able to turn off background shader cache off for specific games until this issue get fixed.also i need background shader cache for specific games like SOT and Halo infinite and turn it on and off while switching games is also wasting Traffic.

**Comment by voorhees3388** [comment]
Same 10.9GB with Enshrouded. pausing and cancelling fixes until steam or PC is rebooted. Other games have this as well but they are small MB vs this one large download each time. I am using btrfs on main drive and added a second btrfs 3TB HDD for games that don't require SSD speeds. I honestly do not remember if this was happening before adding the 2nd drive.

![Screenshot_20240401_192141](https://github.com/ValveSoftware/steam-for-linux/assets/60133119/7c997ab3-0583-42fc-bf33-ad92e82de995)

**Comment by arazilsongweaver** [comment]
I experience this with Warframe all the time on two separate Arch Linux installs. Every time Steam starts up, Warframe triggers a 7.6 GB cache download. I also get the same general behavior (usually with a 1-400 MB download) from every other installed game in my library.

I also think Steam needs a cache management interface (similar to the "Storage" page in settings) that would allow users to see how big each game's cache is and provide an option to disable and/or delete each individual game cache.

**Comment by gregor3000** [comment]
it says "Enabling this feature may slightly increase disk and bandwidth usage." 
my question: Are you insane?

CS2 had a 2GB update each time is tarted steam or the game crashed. others report of 10 GB updates. imagine 100.000 thousands of users downloading for no reason. This has not been resolved. 
This is actually a lot of unnecessary disk (write cycle) and bandwidth usage that is not necessary. why are shaders even downloading and pre-caching for games i am not currently playing and that have received no (0) map updates or updates otherwise at all? why?

**Comment by Zorrototo** [comment]
> it says "Enabling this feature may slightly increase disk and bandwidth usage."
> my question: Are you insane?

This is true, when the feature works as intended, it is obviously not intended to re-download the whole shader cache every time.

**Comment by Hyrules** [comment]
I have this issue as well in arch on Endeavour OS. Same as voorhees3388. My shaders cache for Enshrouded keeps downloading over and over the 10GB of the cache when I restart steam. It happens for other games as well. I don't mind a few megabytes of caching but when it goes over the gigabyte...

**Comment by voorhees3388** [comment]
> I got a 13GB shader cache update for AHIT today, immediately stopped it, switched the game to Proton Experimental, cleared my steam download cache from steams download settings and restarted steam. Then I got the expected 146.5MB shader update... 

My Enshrouded that was downloading ~10GB each reboot was on GE-Proton7-42. Switching to Proton Experimental and clearing steam download cache seems to have solved this and no longer downloads anything upon reboot. 

I was going to try without clearing download cache but forgot, if anyone gets a chance I wonder if this is a necessary step.

**Comment by gregor3000** [comment]
what does proton have to do with it? large "shaders update" are present on Linux native games. CS2 is native and so are L4D, Half life 2, GarrysMod... some of them have a few 100 MB update but they are daily. why? then they re-download the updates when you restart steam or PC during the day and CS2 does it even after crash.

**Comment by sophie-kuehn** [comment]
Afaik Proton don't has anything to do with this. The download of shader pre-cache files is a Steam application feature. That's why this topic is not in the Proton repository. I expect this has something to do with false positive checks for changes or missing files.

**Comment by TheWiFiNerds** [comment]
I hope this one gets picked up for a fix soon. The 40MB updates a few of the games get don't bother me, but the sometimes daily or more than once a day 2GB CS2 downloads followed by shader pre-caching is brutal.

**Comment by kaarelen** [comment]
Same thing happening with with bunch of games, including BG3 (biggest one)

**Comment by gregor3000** [comment]
> I hope this one gets picked up for a fix soon. The 40MB updates a few of the games get don't bother me, but the sometimes daily or more than once a day 2GB CS2 downloads followed by shader pre-caching is brutal.

i turned off shader precaching updates. wish this could be done per game title. 
in any case the result for CS2 is slightly longer map load times (not by much). occasionally it can not load the map - specific shader is not loading (work around: exit game, load game, then map loads fast).

as for other games in my library i am not even sure why shaders are being downloaded, since there are no updates for them.

**Comment by ipaqmaster** [comment]
This has started happening to me this week. Starting Steam immediately validates 1.88 GB of CS2 very slowly despite this NVMe.

ZFS reports no READ/WRITE/CKSUM errors on this drive so I'm not sure what makes Steam do this.

**Comment by hatecode0110** [comment]
There is only one way: delete steam, and waiting for the bug fix.

**Comment by arcangel2p** [comment]
I followed a suggestion on Steam Deck's subreddit. It was to set downloads on a schedule, like 5-6 AM. This way no more automatic cache are downloaded, only when you manually update or start a game.

It's not a solution, but while Valve decides if it will fix this some day (or century...), it helps.

**Comment by Zorrototo** [comment]
Indeed there are workaround, but it doesn't fix the core issue. I'll ask again, what would you need @kisak-valve from us to help more to debug the situation?

**Comment by MechMK1** [comment]
Same happens to me as well, where Steam downloads "updates" for every single game I have installed. Every time I boot any game, I need to compile shaders again. It is annoying.

I also second the plea to please tell us what information is necessary to provide to debug and fix this.

**Comment by xemulat** [comment]
I have exactly the same issue. CachyOS + Nvidia.

![image](https://github.com/ValveSoftware/steam-for-linux/assets/98595166/2cfe7878-52c5-4536-a4a8-a38955e631ed)

**Comment by hatecode0110** [comment]
> @hatecode0110 are you planning to live so long?

After you ❤️

**Comment by Chaman-veteran** [comment]
@drmmwlt could you please lock comments for this issue for now, as to allow people to still follow it without being spam?

**Comment by UnconnectedBedna** [comment]
I have been following these kinds of threads for a long time, because this has been happening for more than a year, both on Manjaro and on Arch, so this is def a steam thing.

What I have identified happening, that I could not find mentioned here in the thread is that yes, these are shaders, but they ONLY download, there is no writing to disk, since the cache already contains the shaders, you can even see this in the steam client because the disk activity is 0.

By guess is it "validates" by downloading ALL THE SHADERS in your entire library, then compares them to what you have cached, and if no change, no writing to disk.

What made me come to this conclusion?
Since steam does NOT redownload after closing and opening it, but it DOES after a reboot, I tend to believe steam puts the downloads somewhere on my system where it is only kept in memory or temp memory (swap).

PLEASE steam, fix this. There has to be a way to use for example a CRC check on the shaders instead of downloading the entire thing. Shaders in some games can be VERY big.
I have to "download" Gibibytes worth of data every reboot.

**Comment by Zorrototo** [comment]
Of course this is a Steam bug.

>My guess is it "validates" by downloading ALL THE SHADERS in your entire library, then compares them to what you have cached, and if no change, no writing to disk.

That's not how it is supposed to work Bredna (Hi there :D) It is supposed to do checksum and/or versioning. I maybe will investigate on this write thing you're talking about though, maybe it can lead somewhere. In the past years they fixed this issue which was most backend issues if I recall correctly, they fixed things on the servers side, maybe it is still the same again.

On my side, I fixed it, because as yourself I had to re-download GBs of useless data everyday and I can't afford it with my 10Mb DSL: ![image](https://github.com/ValveSoftware/steam-for-linux/assets/17416415/362098f0-211f-477b-916b-a5a895a0b5df)

This is a real shame, to not be able to benefit from a good feature. Also a shame nobody from Steam is actively involved in this issue here.

**Comment by UnconnectedBedna** [comment]
> That's not how it is supposed to work Bedna (Hi there :D) It is supposed to do checksum and/or versioning. I maybe will investigate on this write thing you're talking about though, maybe it can lead somewhere. In the past years they fixed this issue which was most backend issues if I recall correctly, they fixed things on the servers side, maybe it is still the same again.

Hey buddy! :)

I should probably also add that my steam installation is on btrfs subvolume, separate home also on btrfs subvolume.
But my steam library is in ext4 on a partition/disk of its own, that might actually have unforeseen consequences since "Steam Common Redistributables" refuse to move from my steam folder on my home.
I honestly have not investigated WHERE the data goes, I just assumed it goes somewhere kept in memory since it only redownloads after reboot.

It is sad devs do not fix this, or even acknowledge the existence of the bug, but I also understand after reading this thread and many before them, that a dev might not want to intervene with comments.

This is supposed to be a bug report, not a discussion. xD (I say while half my post is me ranting) :heart: 

Thank you Valve for throwing all that money into proton, without that we would most likely not even have this conversation, because there would not be any gaming on linux. xD

**Comment by Zorrototo** [comment]
By actively involved I mean someone could ask for specific things to help debug the issue, some way of having hope this is looked at and worked on. We could throw all the logs we can gather but too much information is sometimes really too much to process. To be fair it is maybe just a very low priority issue as it doesn't break being able to play.

**Comment by JohnTheCoolingFan** [comment]
> I should probably also add that my steam installation is on btrfs subvolume, separate home also on btrfs subvolume.
But my steam library is in ext4 on a partition/disk of its own, that might actually have unforeseen consequences since "Steam Common Redistributables" refuse to move from my steam folder on my home.

I have the same happening on my end, but everything is on the same btrfs partition but in multiple subvolumes, and it's also in flatpak. I originally though that this may be an issue of me sharing a library I had on my HDD raid0 across two different linux installs, but it might be just that the timing was right and the bug started to appear when I set that up. But it still happens to me. Quite annoying, but not as bad with a 100MBit connection.

**Comment by JohannesKuehnel** [comment]
Don't worry, this is neither btrfs, flatpak nor raid0 related (not saying you said that, just want to make it clear).
I use neither of them (Arch on single ext4 disk with Steam from Arch repo) and have the same issue.

**Comment by UnconnectedBedna** [comment]
I dug a tiny bit deeper, they DO seem to write to disk after all.

Rebooted, started steam and used `mmin` with `find` and 
`...../SteamLibrary/steamapps/shadercache/<app-id>/fozpipelinesv6/`
seems to be used.

And browsing the files/directories in there seems to be a pretty nice mess with a ton of cache directories, multiple versions of .foz files everywhere with the same name, or just the same file with a number added to it, or at least nothing in there makes sense to me.

**Comment by beach-89** [comment]
> This is still happening almost every reboot like the comment above says. It will download the entire shader cache, then redownload it 5 min later after a restart. Is there any logs or anything that can be provided to help elevate or diagnose this?

This bug is no longer happening for me after the latest steam client update. I tried to restart a few times in a row, and I didn’t get any shader downloads.

**Comment by shelterx** [comment]
It's not fixed, my system uptime is 5 days and steam's been running for ~24h, during that time it's been downloading shaders for the same game twice. I'm going to have to turn shader downloads off because it takes like 30 minutes to process them and tbh I notice no difference in the game I'm playing anyway.

**Comment by ojsl1** [comment]
If you're gonna spam it doesn't work/works at least include your steam version...

I'm on steam beta 1716072097 and haven't come across this anymore for a week now.

**Comment by shelterx** [comment]
Steam Beta Branch:  Steam Beta Update
Steam Version:  1715979898
Steam Client Build Date:  Fri, May 17 10:45 PM UTC -08:00
Steam Web Build Date:  Thu, May 16 9:36 PM UTC -08:00

**Comment by UnconnectedBedna** [comment]
What version is in use is less relevant here since this has been going on for a LONG time.
If a patch were to be referenced to this bug, yes, the latest version would be required for testing.

It is still going on, but to a tiny bit less extent it seems.

BUT:
Yesterday I started playing V rising again after 1.0 release, and no shaders were available for my system so it compiled it for about 20minustes using 100% all my 24 cpu cores. After game it uploaded the whole thing to steam.
You are welcome steam community, I am happy to contribute. :)

What I DO NOT WANT after compiling for that long, is to be forced to DOWNLOAD MY OWN SHADERS (1.2GiB) after I rebooted and started steam again.
As usual, downloading gibibytes of data, but "writing to disk" stays at 0. :(

**Comment by shelterx** [comment]
Mmm V rising is the game I'm playing too currently. I've had it wanting to process shaders 3 times now. Not sure if it actually downloaded something  though.

**Comment by jandd** [comment]
I still see downloads of > 9 GB for Enshrouded (almost?) every day. I'm using the Steam beta client on Debian testing.

**Comment by migrgh** [comment]
Same here download fallout new vegas and next day download over 350,8MB. 
I don't even play the game or change anything. 

- Fedora 40 steam btrfs raid1. 
- `/dev/nvme1n1p3 on / type btrfs (rw,relatime,seclabel,ssd,discard=async,space_cache=v2,subvolid=5,subvol=/)`

**Comment by SadmL** [comment]
Steam Stable 1716584667
Client Build Date: Fri, May 24 23:48 UTC +03:00
Web Build Date: Fri, May 24 23:31 UTC +03:00

Warframe redownloads 12.3 GB every day, 2 month ago it was ~10 GB
![изображение](https://github.com/ValveSoftware/steam-for-linux/assets/30765190/7724e1af-0a73-4749-bc6c-d4532070a8d5)

Also I have 14 GB of shaders cache
![изображение](https://github.com/ValveSoftware/steam-for-linux/assets/30765190/1122f79c-adb9-4219-9f89-c436f15da28e)

On screenshot it still shows ~26 GB, but I manually deleted it
![изображение](https://github.com/ValveSoftware/steam-for-linux/assets/30765190/19151345-6618-4696-b48d-a22ff8dccdfa)

**Comment by dhruv8sh** [comment]
I understand there is less focus on Linux as compared to Windows, but this is horrendous.
Apart from this, it also happens with "Processing Vulkan shaders", which takes a long long time.

**Comment by kaarelen** [comment]
@dhruv8sh if this issue was an infant - at this point, it is just a few weeks from sitting by himself and pronouncing his first "mama" word.

**Comment by Zorrototo** [comment]
> Apart from this, it also happens with "Processing Vulkan shaders", which takes a long long time.

Processing shaders is expected, this is not an issue. Enable background shader processing for the shaders to be processed in the background.

**Comment by dhruv8sh** [comment]
> Processing shaders is expected
Yes it is expected, but not every time I restart my system.
> Enable background shader processing for the shaders to be processed in the background.
I am aware of the feature, I just don't want to enable it.

**Comment by eode** [comment]
> > Apart from this, it also happens with "Processing Vulkan shaders", which takes a long long time.
> 
> Processing shaders is expected, this is not an issue. Enable background shader processing for the shaders to be processed in the background.

It's clear you have not actually read this issue.

The shaders reprocess and redownload almost every time steam starts.  For many players, this boils down to gigabyte+ downloads, usually for multiple games.  This is not normal.  It is not efficient.  It is not good, and for people with less wealth or who don't have gigabit internet, it's a blocker on this entire feature-set.

Worse, if the person isn't tech-savvy enough to realize that they can disable the feature, it's just a blocker for using some games entirely.

Disabling shader precaching is the closest users have to a solution on this - and for all I know, that's not just downloads, that's all shader caching.  When the feature is disabled, lots of space is cleared - and it appears not to build up again locally.

**Comment by UnconnectedBedna** [comment]
> It's clear you have not actually read this issue.

Now, there is no need to get personal, but I happen to know this person is very aware of what the issue is.

And I also want to clarify, just like him that the processing is NOT the issue here, it is the constant re-downloading of the shaders is.

> Disabling shader precaching is the closest users have to a solution on this - and for all I know, that's not just downloads, that's all shader caching. When the feature is disabled, lots of space is cleared - and it appears not to build up again locally.

Also not a solution, but for people who have very slow internet connection, yes, it might be the only workaround as of this moment.
That also means the games might run like complete a** since the shaders have to be compiled live on the hardware while playing.

**Comment by I-OutOfGravity-I** [comment]
same problem. arch cs2 needs new shaderzz every time i open steam

**Comment by radityaargap** [comment]
Must be related to #10906. Games try to download shader every launch, but since pre-shader chache is turned off, it gives disk write error.

**Comment by Fossiel** [comment]
_(Linux Mint, .deb Steam client)_

I've found an inelegant, temporary work-around:
- Set the game to download updates at launch only
- Launch the game from your library and then immediately switch to the download page
- Pause the download
- If the game launches before you can pause, simply hit the super-key to switch back to Steam and pause the download. I don't have a Steamdeck so I don't know how you would do this on the deck.

* If you pause the download and the game doesn't launch, just launch it again without releasing the pause.

Unfortunately, this comes with a major flaw. I'm not sure what will happen when the game pushes an actual update. I'm still experimenting but hopefully, the game will refuse to start when this happens.

**Comment by YoshiRulz** [comment]
I think this has been present for multiple years now. At least with the latest client you can quit and restart Steam and the phantom updates disappear. Valve, please fix.

**Comment by SadmL** [comment]
https://github.com/ValveSoftware/steam-for-linux/issues/10285#issuecomment-2132078830

Looks like in Steam Beta 1719967667 it was fixed. Now Warframe downloads ~40mb, writing on disk ~450mb

**Comment by SadmL** [comment]
https://github.com/ValveSoftware/steam-for-linux/issues/10285#issuecomment-2212325490

Well... It's not. Steam Beta 1720650399
Downloading 12.5Gb, writing 12.9Gb.

**Comment by cosmic-zip** [comment]
I the same problem, borderlands, cyberpunk, cs2. Every day 1.7Gb download, i think this s*** are going to kill my SSD, anyway, the only solution for now are disable vulkan shades

**Comment by Zorrototo** [comment]
I don't see any issue anymore nowadays by having this feature disabled. Maybe all my game I play are working well without a pre-compiled shader cache for the whole game, or maybe Proton is good enough nowadays to just compile shaders on the fly without stuttering.

**Comment by creeper82** [comment]
I've got the same issue on Arch. It's unbelievable that this thread is so old and nothing happened

**Comment by UnconnectedBedna** [comment]
> I don't see any issue anymore nowadays by having this feature disabled. Maybe all my game I play are working well without a pre-compiled shader cache for the whole game, or maybe Proton is good enough nowadays to just compile shaders on the fly without stuttering. Just let go people, it will be fine, just try.

Try activating again.
Not sure WHAT they changed, but it is definitely no longer showing the same tendencies on my system.
At least I have not changed anything on my side, but one day the daily gb downloads just stopped...

**Comment by Zorrototo** [comment]
Honestly I'm done with Shader Pre-Caching :)
It was fun while it was needed, but seriously it's good now I don't know, maybe it is since I switched from GTX 1060 to a more powerful RX6800 AMD card, but I don't see the difference now.

**Comment by kaarelen** [comment]
@Zorrototo shader compilation takes your cpu, not gpu

**Comment by Zorrototo** [comment]
> @Zorrototo shader compilation takes your cpu, not gpu

But having a better graphics card, especially from a GTX 1060, a card from 2016, is surely doing something in the equation, as said I have no issue now with that card without shaders pre-caching. CPU did not change.

Anyway my point was now, it doesn't seem these feature is a crucial point to have a good Proton experience, and maybe the card helped.

**Comment by YoshiRulz** [comment]
To add to the anecdata, I disabled shader precaching after https://github.com/ValveSoftware/steam-for-linux/issues/10285#issuecomment-2234265518 and haven't seen a phantom update even after several reboots. I'd initially dismissed that as the cause since *half the updates didn't have the shader label*.

There's certainly *something* wrong with precaching—1.7 GiB of shaders? The game itself isn't even that big.
![screencap](https://github.com/user-attachments/assets/99bace1c-be98-49cc-be8b-90ca792b90d8)

**Comment by Zorrototo** [comment]
That's expected, if you disable Shader Pre-Caching feature, then you'll not suffer from redownloading all shader caches of your game. When I was talking about disabling it and not having any issue, I was talking about having no issue in the game, no stuttering when compiling shaders on the fly (opposite to having Shader Pre-Caching enabled where shaders are pre-compile before you even start the game).

**Comment by kaarelen** [comment]
@Zorrototo if pre-caching is disabled - shaders will compile at runtime. If your CPU can handle it and the bottleneck stays on the GPU - you won't notice anything. That's it. Hope i've clarified it a bit) Have a good day!

**Comment by Zorrototo** [comment]
Yes I know, compiling shaders is done on the CPU. But my point, again, is that it doesn't seem to be a burden in Proton anymore. AND additionally, maybe the new AMD card helped in the process. A few years ago games were locking hard will compiling shaders, now it seems it is transparent.

**Comment by Summanis** [comment]
Can confirm this still happens. 9.5GB daily Enshrouded download.

**Comment by Zorrototo** [comment]
Did the people with the issue try to simply disable the whole functionality and see if they experience issues? as I already said earlier, it doesn't seem to be an issue now without Shader Pre Caching enabled. Maybe you'll find peace by stopping using the feature.

**Comment by SadmL** [comment]
I've tried disable it. Warfame starts shuttering very much.

**Comment by dhruv8sh** [comment]
Can confirm CS2 fps dropped from 144(vertical sync) to 70-80(vertical sync), same settings after disabling the feature.

**Comment by Zorrototo** [comment]
OK, I should have a try to compare and see if we have same experience.
I guess it depends on the game.

PS: enabling V-sync will force it to drop to half your refresh rate if it can not sustain the refresh rate. I understand about stuttering appearing, but halving your FPS makes no sense here, it is caused by V-sync.

**Comment by Summanis** [comment]
> Can confirm CS2 fps dropped from 144(vertical sync) to 70-80(vertical sync), same settings after disabling the feature.

I just ran community benchmark map and got the exact same results that I did with the shader pre-caching turned on. 

> Did the people with the issue try to simply disable the whole functionality and see if they experience issues?

Yes, disabling shader pre-caching does fix the daily shader redownloads. Its a cool feature that I hope gets fixed, but I'd rather not be downloading 10+GB a day that doesn't even get written to disk because its already there.

**Comment by dhruv8sh** [comment]
> I just ran community benchmark map and got the exact same results that I did with the shader pre-caching turned on.

Did you clear the caches first before testing? If it's there of course the performance is going to be identical.

> enabling V-sync will force it to drop to half your refresh rate if it can not sustain the refresh rate.

Exactly, it looks like the cpu is working harder to compile them in the background. Could be something else, but in CS2 this result was pretty consistent for me.

144fps when I let it process shaders, else I'm playing at 80.

**Comment by Summanis** [comment]
> Did you clear the caches first before testing? If it's there of course the performance is going to be identical.

I'm not sure how to clear the cache. Steam deleted its "precaching" content and /.cache/mesa_shader_cache is empty, though /.cache/mesa_shader_cache_db isn't.

**Comment by Zorrototo** [comment]
It is `~/.local/share/Steam/steamapps/shadercache` or something like that, where the Shader Pre Caching cache should be, delete the folder if disabling the feature did not delete it all.

**Comment by Summanis** [comment]
Well I'm definitely confused now. FPS went way down after deleting the shadercache, even if I ran the game for several minutes hoping it would recompile its own shaders. After enabling shader precaching and downloading shaders for cs2, the FPS actually got WORSE. I restarted my PC and it went back to what it was originally. I thought that maybe I should try running around in game to "rebuild the shader cache" and then restarting my PC, but this time after deleting the shadercache folder the FPS stayed the same, even post reboot. I probably did something wrong but I don't know what.

AVG 380 1% low 180 with shader precaching (ran yesterday, small update since)
AVG 255 1% low 127 after deleting shadercache folder
AVG 218 1% low 108 after redownloading the shader cache
AVG 377 1% low 178 after rebooting

**Comment by markau** [comment]
I'm getting daily 15GB updates to Overwatch 2 for the past week or so. 

Operating System: Ubuntu 24.04.1 LTS                
Kernel: Linux 6.8.0-47-generic

**Comment by UnconnectedBedna** [comment]
> > I'm getting daily 15GB updates to Overwatch 2 for the past week or so.
> > Operating System: Ubuntu 24.04.1 LTS Kernel: Linux 6.8.0-47-generic
> 
> I have the same problem. any solution?

No, but there are 2 workarounds that I know of.
One is mentioned earlier, by disabling shader caching completely and removing the cache.

But a better solution I have found is to set every game (I have not found a way to do this globally) to (rightclick on game) properties > updates > automatic updates: only update this game when I launch it.

The reason the download happens is when the games are verified, it also downloads something for some reason.
By disabling the autoupdate, that behavior stops.
And when launching the game with this setting, there is no verification of files (the bug), it only updates in case there actually IS an update for the game.

**Comment by dom1torii** [comment]
> > > I'm getting daily 15GB updates to Overwatch 2 for the past week or so.
> > > Operating System: Ubuntu 24.04.1 LTS Kernel: Linux 6.8.0-47-generic
> > 
> > 
> > I have the same problem. any solution?
> 
> No, but there are 2 workarounds that I know of. One is mentioned earlier, by disabling shader caching completely and removing the cache.
> 
> **But a better solution I have found recently** is to set every game (I have not found a way to do this globally) to (rightclick on game) properties > updates > automatic updates: **_only update this game when I launch it._**
> 
> The reason the download happens is when the games are verified, it also downloads something for some reason. By disabling the autoupdate, that behavior stops. And when launching the game with this setting, there is no verification of files (the bug), it only updates in case there actually IS an update for the game.

Your solution is really good. Thanks

**Comment by DerCommander323** [comment]
I think I'm encountering the same thing for CS2 mainly. However what makes it much worse is that instead of on Steam startup, CS2 starts "validating" when I _start the game_, which causes game launch to take like 5 minutes while also freezing up my entire system for short moments constantly. It's extremely annoying to have to start the game, close it, wait for Steam to do... _something_ for like 10 minutes straight and then finally be able to actually play the game...

This is on a Steam Deck Oled with vanilla Arch Linux installed, running Hyprland.

**Comment by shitfish** [comment]
I’ve been having an issue with CS2 where I constantly need to verify the game files every time I launch it. After troubleshooting, I discovered something interesting:

If I launch the game using a shortcut (whether from the taskbar or desktop), the game always requires me to verify the files before starting.

However, if I launch CS2:

By right-clicking on Steam in the taskbar and selecting the game from the context menu,
OR directly from the Steam library,
...the game launches fine, without needing to verify files.
Here’s the key difference I noticed: When I launch it through Steam, it does a quick cloud sync just before starting the game, which doesn’t happen when launching via the shortcut (even if the shortcut points to the cs2.exe file).

I suspect the problem is that the shortcut doesn’t include some launch parameters or doesn’t initialize something critical, like the anti-cheat system, which Steam seems to handle properly.

**Comment by bushibot** [comment]
Wow still a problem, trying work arounds but really, its's been years?

**Comment by creeper82** [comment]
Yup. I'm also amazed how it hadn't been fixed yet. I gave up and disabled the shader cache overall

**Comment by Yilmaz4** [comment]
Having the same issue with BeamNG.drive, every time I start Steam, it says "validating" and downloads an arbitrary size of data, sometimes 700MB sometimes 3GB. This drained my mobile data in a matter of minutes when I opened Steam and forgot about it while I was without Wi-Fi!

**Comment by dom1torii** [comment]
Had this issue for a long time, but now it started causing steam and game to fully freeze when its on 100% and i need to restart them by killing processes. Now even "Only update this game when i launch it" doesn't help, because it will just start downloading when i open game and freeze the same way. Only fix for this i found is fully disabling shader cache.

**Comment by 404nnotfoundddd** [comment]
I have same problem on linux mint. Steam validates CS2 everyday.

**Comment by error161** [comment]
I have multiple games that do this. I get ~8GB Rocket League updates repeatedly (like.. every other day). I get PoE2 updates constantly too. This is even without restarting the system or client. Downloaded an 8.1GB of RL shaders Friday, and that same exact 8.1GB today.

**Comment by grishatop1** [comment]
valve pls fix

**Comment by RoanBrand** [comment]
Valve please pls pls fix

**Comment by Matheos96** [comment]
Same issue. RL. Quite frustrating when you need to get on 2v2 tournament starting in 1min and you miss it because you just turned on your pc :(

**Comment by Zorrototo** [comment]
It has always been there.

**Comment by n1flh31mur** [comment]
> I have same problem on linux mint. Steam validates CS2 everyday.

Same problem on Arch linux.

**Comment by Notmare** [comment]
I switched to Linux last week and didn't notice an issue until I installed Enshrouded.  At first I thought the game was actually updating regularly, but now I understand that it's just validating the cache every time I start Steam.  I'm trying the proposed workaround of only updating on game launch, but I do wish that I didn't have to do that.

**Comment by Soulprayer79** [comment]
> I switched to Linux last week and didn't notice an issue until I installed Enshrouded. At first I thought the game was actually updating regularly, but now I understand that it's just validating the cache every time I start Steam. I'm trying the proposed workaround of only updating on game launch, but I do wish that I didn't have to do that.

Deactivate shader precaching. you can find it within steam -> settings -> Downloads -> shader pre-caching

If you have an average PC (for nowadays) you should have no impact on your gaming experience.

**Comment by UnoYakshi** [comment]
I have this problem every time I quit the game (Enshrouded), for some reason.
![Image](https://github.com/user-attachments/assets/4b3d588d-a565-47f7-b6be-ecfaee29c2ba)

**Comment by KarLitoDE** [comment]
> Did I completely miss this setting earlier or did they implement this since I made the comment about disabling auto-updates for each and every game nov last year?
> 
> No matter what, do this in settings > downloads as a workaround for this bug:
> 
> ![Image](https://github.com/user-attachments/assets/547d5ad6-6f03-4e41-9970-ae33779b4989)

But do i have to download the 9 gigs when i start the game?

____
My Problem since couple of days is with a 9gb Pre-Shader Cache download for Rocket League

![Image](https://github.com/user-attachments/assets/285d0373-7b0b-4c75-8763-504ce1341110)
Everyday after starting Steam I have to download the same size of "Pre-Shader Cache".

**Specs**:
OS: Zorin x86_64 
Host: 5943 
CPU: AMD Ryzen 9 5900X (24) @ 3.7GHz 
GPU: NVIDIA GeForce RTX 3080 Lite Ha 
Memory: 5,70GiB / 31,26GiB (18%) 
GPU Driver: NVIDIA 565.77 
**Steam**: 
**Steam Beta Branch:  Stable Client
Steam Version:  1738026274**
Steam Client Build Date:  Tue, Jan 28 01:50 UTC -08:00
Steam Web Build Date:  Tue, Jan 28 01:44 UTC -08:00
Steam API Version:  SteamClient021

**Comment by UnconnectedBedna** [comment]
I'll post it again:

There are two workarounds known right now.

1. Enable "only update on launch" (I have not had any trouble since doing that)
2. Disable pre-caching shaders completely

I prefer option 1, others option 2. Both works for stopping the downloads.
Up to you to decide if you want pre-cahcing or not.

**Comment by n1flh31mur** [comment]
> Enable "only update on launch" (I have not had any trouble since doing that)

Enabling this option cause that game won't update on steam launch, but will update on game launch, and at the end of update game will freeze completely.

So I can't say this can be workaround as a "disabling pre-caching" 🤔

**Comment by UnconnectedBedna** [comment]
> > Enable "only update on launch" (I have not had any trouble since doing that)
> 
> Enabling this option cause that game won't update on steam launch, but will update on game launch, and at the end of update game will freeze completely.
> 
> So I can't say this can be workaround as a "disabling pre-caching" 🤔

No it doesn't. I might SAY there is an update, but when you press "update", in 9 out of 10 times it will make the crc check (or whatever it does) towards your cached files locally and compare with what wants to be downloaded and then skips it all because the check matches on both sides.
**THAT is the bug.** If steam automatically updates immediately at start, the crc check does not seem to be performed correctly, hence leading to constant re-downloads.

If it downloads it means either of two things:
- There actually IS a game update
- There are more efficient shaders compiled and uploaded to steam, hence download them (happens to me with brand new released games)

**Comment by Paco-4272** [comment]
I'm having the same problem.
I switched to Linux a few days ago and I'm playing Enshrouded. I have to download 10.3 GB every time I want to play, and I don't have fiber, so it's taking a long time...

**Comment by raelite** [comment]
for me; l4d2.
did tranfer it from a external drive to my laptop, this is where this issue had begone.
did try a reinstall etc.. somewhere there might be a left-over file triggerin' it, maybe?
fedora, plasma.

**Comment by gromkaaa** [comment]
![Image](https://github.com/user-attachments/assets/98217a2c-d628-4ef2-aa65-4c3d7c32344b)

Same here but it doesn't seem to be a bug it is shader pre-caching update. Described here:

![Image](https://github.com/user-attachments/assets/73ba34d4-3337-403e-a5d1-cb8c79b748f1)

**Comment by dbptwg** [comment]
Disabling shader pre-caching resolved this for me when the issue was small verifications on game close.

**Comment by phil-de-choc** [comment]
I have the same problem with Counter-Strike 2:
![Image](https://github.com/user-attachments/assets/dafde5a1-8bd0-4461-be9f-8f376fbca347)
When there is no true game update, the shaders are re-cached each day, causing a ~2.5 Gb download when I start Steam.  I cannot believe this is intended behavior, it makes no sense.

My system:
OS: Debian 12.10 Linux-x86_64
Kernel: 6.1.0-32-amd64
Steam Beta Branch:  Stable Client
Steam Version:  1745876290
Steam Client Build Date:  Mon, Apr 28 5:09 PM UTC -08:00
Steam Web Build Date:  Fri, Apr 25 5:04 PM UTC -08:00
Steam API Version:  SteamClient022
GPU: NVIDIA GeForce RTX 2070 SUPER
NVIDIA Driver Version: 535.216.01

BTW I tried disabling shader pre-caching, but then the shaders are processed when the maps load, and often the shader compilation stops before completion, rendering the game unresponsive and leaving me with no mouse control when I switch window.
![Image](https://github.com/user-attachments/assets/2d8b852b-dc26-4496-82fc-a231f185570f)

Valve plz fix.

**Comment by UnconnectedBedna** [comment]
Witch is why I say the ~~solution~~ fix I came up with is better if you don't have very powerful hardware where your computer is able to recompile shaders at runtime without you noticing it.

Set steam to "Only update at games launch" and you get rid of the problem, the shades will not re-download, only updates will be applied (if there are any) when you start the game.

![Image](https://github.com/user-attachments/assets/cf64eb3b-53c5-4faf-8789-042e1f806e69)

**Comment by phil-de-choc** [comment]
> Witch is why I say the ~solution~ fix I came up with is better if you don't have very powerful hardware where your computer is able to recompile shaders at runtime without you noticing it.
> 
> Set steam to "Only update at games launch" and you get rid of the problem, the shades will not re-download, only updates will be applied (if there are any) when you start the game.

I tried the option "Only update at games launch", with Shader Pre-caching enabled, and now the Shader download is after I leave the game, the first time of the day, instead of when Steam starts, which isn't that much better.  Even when there isn't any game update.  I still believe this should be addressed by Valve.

**Comment by blipk** [comment]
Billions of dollars in loot boxes and you can't hire some people to fix basic issues, why valve?

**Comment by Zorrototo** [comment]
Probably because it is not basic issue.

**Comment by detaer** [comment]
This happens to me quite frequently on ubuntu.

**Comment by eode** [comment]
Hey guys -- for some of you, this may fix the issue.

* If you're running into this issue
* and your cache (or library) is on a different drive / partition than the default folder
* and you have enough room for the cache on the drive with the default steam folder

then try this:

```bash
# enter the non-default library
cd /<path to non-default steam library>/steamapps/
# Optional: copy over all shadercache data to the default library's shadercache
cp -a shadercache/* /<path to default steam library>/steamapps/shadercache/
# backup or delete the non-default library's shadercache
mv shadercache shadercache.backup   # you may want to delete this folder instead - it's huge.
# link the default library's shadercache into the current folder (the non-default library)
ln -s /<path to default steam library>/steamapps/shadercache  
```
This makes it so that when the non-default library is used for play or downloads, it uses the default library's shadercache.

If you run into problems, delete the default `shadercache` folder so that all is re-downloaded.  If you still run into problems, delete the `shadercache` links (which will cause Steam to just re-create the folders and go back to default behavior).  Note that deleting these folders will cause steam to have to re-download the shaders.  ..but, that's already happening, so...

**Comment by zexi** [comment]
Assuming the Steam directory is at `~/.steam/steam`, after the first pre-shader compilation is complete, rename `~/.steam/steam/ubuntu12_64/fossilize_replay` to `fossilize_replay.bk`, and then create a dummy Bash script. 

This might prevent the shaders from recompiling every time?

```bash
mv ~/.steam/steam/ubuntu12_64/fossilize_replay ~/.steam/steam/ubuntu12_64/fossilize_replay.bk
echo 'exit 0' >  ~/.steam/steam/ubuntu12_64/fossilize_replay
chmod a+x ~/.steam/steam/ubuntu12_64/fossilize_replay
```

**Comment by KebabLord** [comment]
bump, issue is still occuring on my end, everytime i restart steam client its downloading 1.3GB content for CS2
deleting "~/.local/share/Steam/steamapps/shadercache" and additional disk's library "/disk2/SteamLibrary/steamapps/shadercache/*" doesn't resolve the issue. Along with this issue everytime i launch a game it downloads the vulkan shaders again which takes forever.

```
Distro: Debian 13
Package: steam:i386
Version: 1:1.0.0.83~ds-3
```

**Comment by UnconnectedBedna** [comment]
> bump, issue is still occuring on my end, everytime i restart steam client its downloading 1.3GB content for CS2 deleting "~/.local/share/Steam/steamapps/shadercache" and additional disk's library "/disk2/SteamLibrary/steamapps/shadercache/*" doesn't resolve the issue. Along with this issue everytime i launch a game it downloads the vulkan shaders again which takes forever.

You can either try to do the symlink thing described in https://github.com/ValveSoftware/steam-for-linux/issues/10285#issuecomment-3238843367
Or just disable autoupdate as I described in https://github.com/ValveSoftware/steam-for-linux/issues/10285#issuecomment-2852066896

I have not had this issue since I set all my games to "only update at game launch", which indicates to me that this is some kind of race condition. All I know is the issue is gone after doing that, and if it is a race condition, symlinks will not fix that but waiting a few seconds does, ie not start updating immediately on steam client launch.

For me this changes close to nothing, because I just click the "manage downloads" (at the bottom of the steam client) and pull up the games I think I will play it in the near future and they will be updated immediately. But your opinion if this is ok or not might be different.

**Comment by eode** [comment]
man, it's crazy that this issue is still around.

**Comment by markau** [comment]
This issue drove me nuts on 2 different distros. The answer was as simple as:

1. install GE-Proton (and set the compatibility for the game to use this)
2. PROTON_LOCAL_SHADER_CACHE=1 launch option
3. turn off Downloads -> shader pre-caching

**Comment by phil-de-choc** [comment]
> This issue drove me nuts on 2 different distros. The answer was as simple as:
> 
>     1. install GE-Proton (and set the compatibility for the game to use this)
> 
>     2. PROTON_LOCAL_SHADER_CACHE=1 launch option
> 
>     3. turn off Downloads -> shader pre-caching

Using GE-Proton (or any Proton version) with Valve Anti-Cheat (VAC) will not get you banned, but it will prevent you from joining official VAC-secured servers. VAC generally fails to initialize or authentic properly through compatibility layers, resulting in an automatic kick or an "unstable/invalid signature" error rather than a penalty.
